text_extensions_for_pandas/io/watson.py

```diff
@@ -0,0 +1,511 @@
+                offset_arrays.append(raw.offsets)
+                null_count += raw.null_count
+                raw = raw.flatten()
+
+            # TODO handle lists with null values
+            if null_count > 0:
+                continue
+
+            # Convert values to numpy
+            values = raw.to_numpy(zero_copy_only=False)  # string might copy
+            offsets_list = [o.to_numpy() for o in offset_arrays]
+
+            # Compute the length of each list in the array
+            value_offsets = offsets_list.pop()
+            value_lengths = value_offsets[1:] - value_offsets[:-1]
+
+            # Separate the arguments into individual columns
+            for i in range(num_arguments):
+                arg_name = "{}.{}.{}".format(name_split[0], i, name_split[1])
+                arg_lengths = value_lengths[i::num_arguments]
+
+                # Fixed length arrays can be sliced
+                if not is_nested_list or len(np.unique(arg_lengths)) == 1:
+                    num_elements = len(first_list[i]) if is_nested_list else 1
+
+                    # Only 1 element so leave in primitive array
+                    if not is_nested_list or num_elements == 1:
+                        arg_values = values[i::num_arguments]
+                        arg_array = pa.array(arg_values)
+                    # Multiple elements so put back in a list array
+                    else:
+                        arg_values = values.reshape([len(col) * num_arguments, num_elements])
+                        arg_values = arg_values[i::num_elements]
+                        arg_values = arg_values.flatten()
+                        arg_offsets = np.cumsum(arg_lengths)
+                        arg_offsets = np.insert(arg_offsets, 0, 0)
+                        arg_array = pa.ListArray.from_arrays(arg_offsets, arg_values)
+                else:
+                    # TODO Argument properties with variable length arrays not currently supported
+                    continue
+
+                flattened_arguments.append((arg_array, arg_name))
+            drop_cols.append(name)
+
+    # Add the flattened argument columns
+    for arg_array, arg_name in flattened_arguments:
+        table = table.append_column(arg_name, arg_array)
+
+    # Drop columns that have been flattened
+    table = table.drop(drop_cols)
+
+    return table.to_pandas()
+
+
+def watson_nlu_parse_response(response: Dict[str, Any],
+                              original_text: str = None,
+                              apply_standard_schema: bool = False) -> Dict[str, pd.DataFrame]:
+    """
+    Parse a Watson NLU response as a decoded JSON string, e.g. dictionary containing requested
+    features and convert into a dict of Pandas DataFrames. The following features in the response
+    will be converted:
+        * entities
+        * keywords
+        * relations
+        * semantic_roles
+        * syntax
+
+    TODO: add desc of feature options, link to API
```

Comment:
Please add some appropriate links here. This docstring will be part of our Python module's documentation. There should be a link to the main landing page for NLU and to the main documentation page for the APIs.